examples/brats18/pt/validators/brats_validator.py

```diff
@@ -0,0 +1,196 @@
+        self.model = monai.networks.nets.SegResNet(
+            blocks_down=[1, 2, 2, 4],
+            blocks_up=[1, 1, 1],
+            init_filters=16,
+            in_channels=1,
+            out_channels=3,
+            dropout_prob=0.2,
+        ).to(self.device)
+        self.model.to(self.device)
+
+        self.transform_valid = Compose(
+            [
+                LoadImaged(keys=["image", "label"]),
+                EnsureChannelFirstd(keys="image"),
+                ConvertToMultiChannelBasedOnBratsClassesd(keys="label"),
+                Spacingd(
+                    keys=["image", "label"],
+                    pixdim=(1.0, 1.0, 1.0),
+                    mode=("bilinear", "nearest"),
+                ),
+                DivisiblePadd(keys=["image", "label"], k=32),
+                Orientationd(keys=["image", "label"], axcodes="RAS"),
+                NormalizeIntensityd(keys="image", nonzero=True, channel_wise=True),
+            ]
+        )
+        self.transform_post = Compose(
+            [Activations(sigmoid=True), AsDiscrete(threshold=0.5)]
+        )
+
+        # Set dataset
+        self.train_dataset = CacheDataset(
+            data=train_list,
+            transform=self.transform_valid,
+            cache_rate=0.2,
+            num_workers=1,
+        )
+        self.valid_dataset = CacheDataset(
+            data=valid_list,
+            transform=self.transform_valid,
+            cache_rate=0.2,
+            num_workers=1,
+        )
+
+        self.train_loader = DataLoader(
+            self.train_dataset,
+            batch_size=1,
+            shuffle=True,
+            num_workers=1,
+        )
+        self.valid_loader = DataLoader(
+            self.valid_dataset,
+            batch_size=1,
+            shuffle=False,
+            num_workers=1,
+        )
+
+        # Set inferer and evaluation metric
+        self.inferer = SlidingWindowInferer(
+            roi_size=self.infer_roi_size, sw_batch_size=1, overlap=0.5
+        )
+        self.valid_metric = DiceMetric(
+            include_background=True, reduction="mean"
+        )  # metrics for validation
+
+    def local_valid(self, val_loader, abort_signal: Signal):
+        self.model.eval()
+        with torch.no_grad():
+            for val_data in val_loader:
+                if abort_signal.triggered:
+                    return
+                val_inputs, val_labels = (
+                    val_data["image"].to(self.device),
+                    val_data["label"].to(self.device),
+                )
+                val_outputs = self.inferer(val_inputs, self.model)
+                val_outputs = [
+                    self.transform_post(i) for i in decollate_batch(val_outputs)
+                ]
+                self.valid_metric(y_pred=val_outputs, y=val_labels)
+            metric = self.valid_metric.aggregate().item()
+            self.valid_metric.reset()
+        print(f"valid_dice: {metric:.4f}")
+        return metric
```

Comment:
Can this be consolidated to the Trainer?

Reply:
Will update and consolidate together with the Learner update